prow/def.bzl

```diff
@@ -138,19 +140,21 @@ def component(cmd, *kinds, **kwargs):
 #   k8s_objects(name = "plank", objects=[":plank_deployment"])
 #   k8s_objects(name = "deployment", objects=[":hook_deployment", ":plank_deployment"])
 #   k8s_objects(name = "service", objects=[":hook_service"])
-#   k8s_objects(name = "fancy", objects=[":hook", ":plank", ":deployment", ":service"])
+#   k8s_objects(name = "staging", objects=[":hook_deployment", ":hook_service", ":plank_deployment"])
 def release(name, *components):
   targets = {}
   objs = []
   for cs in components:
     for (n, ts) in cs.items():
-      targets.setdefault(n, []).extend(ts)
+      if n == "all":
+        objs.extend(ts)
```

Comment:
just create a `k8s_objects(name=name, objects=ts)` here and you can drop references to `obj`?

Reply:
No that won't work how you are suggesting. We would at best wind up with all the targets from a single component or in reality error out with a "redefining target %s" % name as soon as the second component exists.

```
components["hook"] = {"all": ["hook_deployment", "hook_service"], "hook": ["hook_deployment", "hook_service"], "deployment": ["hook_deployment"], "service": ["hook_service"]}
components["deck"] = {"all": ["deck_deployment", "deck_service"], "deck": ["deck_deployment", "deck_service"], "deployment": ["deck_deployment"], "service": ["deck_service"]}
```

`ts` will be either `["deck_deployment", "deck_service"]` or `["hook_deployment", "hook_service"]` 

And we want `k8s_objects(name=name, objs=objs)`, where `objs` is the union of all alls, which is what happens on L158